unit_test/blas/Test_Blas_gesv.hpp

```diff
@@ -310,87 +235,167 @@ int test_gesv_mrhs(const char* mode) {
 #if defined(KOKKOSKERNELS_INST_FLOAT) || (!defined(KOKKOSKERNELS_ETI_ONLY) && !defined(KOKKOSKERNELS_IMPL_CHECK_ETI_CALLS))
 TEST_F( TestCategory, gesv_float ) {
   Kokkos::Profiling::pushRegion("KokkosBlas::Test::gesv_float");
-#ifdef KOKKOSKERNELS_ENABLE_TPL_MAGMA
-  if( std::is_same< typename TestExecSpace::execution_space, Kokkos::Cuda >::value )
+#ifdef KOKKOSKERNELS_ENABLE_TPL_MAGMA //have MAGMA TPL
+  if( std::is_same< typename TestExecSpace::memory_space, Kokkos::CudaSpace >::value ) { //CudaSpace
     test_gesv<float,TestExecSpace> ("N");//No pivoting
-#endif
     test_gesv<float,TestExecSpace> ("Y");//Partial pivoting
+  }
+  #ifdef KOKKOSKERNELS_ENABLE_TPL_BLAS
+  else { //Only test with BLAS TPL
+    test_gesv<float,TestExecSpace> ("Y");//Partial pivoting
+  }
+  #endif
+#else //not have MAGMA TPL
+  #ifdef KOKKOSKERNELS_ENABLE_TPL_BLAS //but test if having BLAS TPL
+    test_gesv<float,TestExecSpace> ("Y");//Partial pivoting
+  #endif
+#endif
```

Comment:
This looks like the place to assert for runtime errors via `try-catch` statements so that we can cover the new KokkosBlas_gesv.hpp code paths [here](https://github.com/kokkos/kokkos-kernels/pull/1008/files#diff-b5deaec19b2ac15994faa8b755b691080c0f07ea3ef7d21fcee9eca1a20a5dbfR105) and [here](https://github.com/kokkos/kokkos-kernels/pull/1008/files#diff-b5deaec19b2ac15994faa8b755b691080c0f07ea3ef7d21fcee9eca1a20a5dbfR114). For example, see https://github.com/kokkos/kokkos-kernels/blob/develop/unit_test/batched/Test_Batched_BatchedGemm.hpp#L57.

Reply:
@e10harvey I have added the try-catch statements in this unit test file.